clippy_lints/src/consts.rs

```diff
@@ -229,7 +229,18 @@ pub fn constant_simple(lcx: &LateContext, e: &Expr) -> Option<Constant> {
     constant(lcx, e).and_then(|(cst, res)| if res { None } else { Some(cst) })
 }
 
-struct ConstEvalLateContext<'a, 'tcx: 'a> {
+/// Creates a ConstEvalLateContext from the given LateContext and TypeckTables
+pub fn constant_context<'c, 'cc>(lcx: &LateContext<'c, 'cc>, tables: &'cc ty::TypeckTables<'cc>) -> ConstEvalLateContext<'c, 'cc> {
+    ConstEvalLateContext {
+        tcx: lcx.tcx,
+        tables,
+        param_env: lcx.param_env,
+        needed_resolution: false,
+        substs: lcx.tcx.intern_substs(&[]),
+    }
+}
+
+pub struct ConstEvalLateContext<'a, 'tcx: 'a> {
```

Comment:
It does not feel ok to make this public. Even that, I saw other public functions on this lint that are being used on `hir_utils`.

Should this be moved to the `utils` module?

Reply:
I think it's fine. I'm planning a big refactor anyway